src/data_structures/suffix_array.rs

```diff
@@ -19,12 +20,79 @@ use vec_map::VecMap;
 
 use alphabets::{Alphabet, RankTransform};
 use data_structures::smallints::SmallInts;
+use data_structures::bwt::{BWT, Less, Occ};
 
+pub trait SuffixArray {
+    fn get(&self, index: usize) -> Option<usize>;
+    fn range(&self, range: Range<usize>) -> Option<Vec<usize>>;
```

Comment:
I thought this might work (untested):

```
fn range(&self, range: Range<usize>) -> Option<Box<Deref<Target = [usize]>>> {
    if range.end <= self.len() {
           Some(Box::new(self[range]))
       } else {
           None
     }
}
```
What error do you get if you try this?

If that does not work, I would actually prefer to have two methods, ``range_slice`` and ``range_vec``. Then, the not working method can throw an unimplemented error. But I really think that above should work in some way. I might ask the rust community if it fails for you.